next/components/media.tsx

```diff
@@ -1,16 +1,35 @@
+import React from "react";
+
+import { MediaFileAttachments } from "@/components/media--file-attachments";
 import { MediaImage } from "@/components/media--image";
 import { MediaVideo } from "@/components/media--video";
-import { Image, Video } from "@/lib/zod/paragraph";
+import { FileAttachments, Image, Video } from "@/lib/zod/paragraph";
 
 interface MediaProps {
-  media: Image["field_image"] | Video["field_video"];
+  media?:
+    | Image["field_image"]
+    | Video["field_video"]
+    | FileAttachments["field_file_attachments"];
+  priority?: boolean;
 }
 
 export function Media({ media, ...props }: MediaProps) {
   if (!media) {
     return null;
   }
 
+  function isFileAttachments(
+    media: MediaProps["media"]
+  ): media is FileAttachments["field_file_attachments"] {
+    return Array.isArray(media) && media[0]?.type === "media--document";
```

Comment:
At the moment this is the only "media" that has multiple values. 

This function here will single out the file attachments, and render the appropriate component.